src/core-members-group/CMGRedemptionHandler.sol

```diff
@@ -276,13 +274,15 @@ contract CMGRedemptionHandler is CMGHandler, ICMGRedemptionHandler, CirclesTypes
         (uint256 ongoingConversion, address beneficiary) = _expectingConversionReturn();
         // starting branch: receive groupId to initiate redemption
         if (ongoingConversion == 0 && _id == cmGroupId) {
-            (uint256[] memory ids, uint256[] memory amounts) = findCollateral(address(cmGroup), _value, false);
-            // move the cursor forward to cycle through other collateral next time
-            cursor = (cursor + ids.length) % activeCollateralIds.length;
-
-            if (ids.length > 0) {
-                // todo: tstore _data so we can recover it on completion
-                _initiateConversion(_from, _value);
+            _initiateConversion(_from, _value);
+            if (_data.length != 0) {
```

Comment:
this is part of your proposal, correct? the other changes are the fixes from Max's review and I, right?

Reply:
it is part of proposal, mainly it makes testGroupRedemptionFlow pass and operate redemption correctly.